google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcResultSetMetaDataTest.java

```diff
@@ -0,0 +1,512 @@
+    when(connection.getCatalog()).thenReturn("test-database");
+    when(statement.getConnection()).then(new Returns(connection));
+
+    subject = resultSet.getMetaData();
+  }
+
+  private static List<TestColumn> createTestColumns() {
+    List<TestColumn> res = new ArrayList<>();
+    int index = 1;
+    for (Type type : getAllTypes()) {
+      TestColumn.Builder builder = TestColumn.Builder.getBuilder();
+      builder.withName("COL" + index).withType(type).withSize(getDefaultSize(type));
+      if (index % 2 == 1)
+        builder.withNotNull();
+      else
+        builder.withNullable();
+      res.add(builder.build());
+      index++;
+    }
+    TestColumn.Builder builder = TestColumn.Builder.getBuilder();
+    builder.withName("CALCULATED").withType(Type.int64()).withNullableUnknown()
+        .withCalculated(true);
+    res.add(builder.build());
+    return res;
+  }
+
+
+  private static int getDefaultSize(Type type) {
+    if (type == Type.string())
+      return 100;
+    return 0;
+  }
+
+  private static List<Type> getAllTypes() {
+    List<Type> types = new ArrayList<>();
+    types.add(Type.bool());
+    types.add(Type.bytes());
+    types.add(Type.date());
+    types.add(Type.float64());
+    types.add(Type.int64());
+    types.add(Type.string());
+    types.add(Type.timestamp());
+    List<Type> arrayTypes = new ArrayList<>();
+    for (Type type : types) {
+      arrayTypes.add(Type.array(type));
+    }
+    types.addAll(arrayTypes);
+
+    return types;
+  }
+
+  private JdbcResultSet getFooTestResultSet(Statement statement) throws SQLException {
+    List<Struct> rows = new ArrayList<>(4);
+    for (int row = 1; row <= 4; row++) {
+      Struct.Builder builder = Struct.newBuilder();
+      for (TestColumn col : TEST_COLUMNS) {
+        builder.set(col.name).to(getDefaultValue(col.type, row));
+      }
+      rows.add(builder.build());
+    }
+    StructField[] fields = new StructField[TEST_COLUMNS.size()];
+    int index = 0;
+    for (TestColumn col : TEST_COLUMNS) {
+      fields[index] = StructField.of(col.name, col.type);
+      index++;
+    }
+
+    ResultSet rs = ResultSets.forRows(Type.struct(fields), rows);
+    return JdbcResultSet.of(statement, rs);
+  }
+
+  private Value getDefaultValue(Type type, int row) {
+    if (type == Type.bool())
+      return Value.bool(Boolean.TRUE);
+    if (type == Type.bytes())
+      return Value.bytes(ByteArray.copyFrom("test byte array " + row));
+    if (type == Type.date())
+      return Value.date(com.google.cloud.Date.fromYearMonthDay(2018, 4, 1));
+    if (type == Type.float64())
+      return Value.float64(123.45D);
+    if (type == Type.int64())
+      return Value.int64(12345L);
+    if (type == Type.string())
+      return Value.string("test value " + row);
+    if (type == Type.timestamp())
+      return Value.timestamp(com.google.cloud.Timestamp.now());
+
+    if (type.getCode() == Code.ARRAY && type.getArrayElementType() == Type.bool())
```

Comment:
let's refactor this to a larger if statement checking if the code == ARRAY

Reply:
done